src/concepts/config-intro.md

```diff
@@ -44,19 +44,46 @@ A route exists to direct messages on the stream to a desired exit point.
 
 ### When a Route matches
 
-Each route can list conditions (any match) to match entry data streams. Attributes like headers, metadata, source, destination, etc., are used `when` determining the correct `exit` for a message.
-
-#### Path and Method matches
+Each route can list conditions to match entry data streams. Route conditions will match if any of the conditions defined match the data stream (any match). Attributes like headers, metadata, source, destination, etc., are used `when` determining the correct `exit` for a message.
```

Comment:
This is probably true but perhaps misleading as there's also more to it.

Each route is checked in order, first match wins, falling through to the default exit if no other routes match.
A route matches if any of its when conditions match, and within the specific when condition, all parts of that condition must match. So if multiple when headers are supplied for http routing, then all of those headers must match for the specific when condition to match.

It's important to mention route evaluation order with default exit, and also draw the distinction between any-match for when conditions vs all-match for each when condition.

Reply:
Ack.

> It's important to mention route evaluation order with default exit

The page does call attention to order just [above this section](https://docs.aklivity.io/zilla/latest/concepts/config-intro.html#routes). Do you think this is enough or we still need to mention it a second time?

Reply:
Ah apologies, I missed that because it wasn't part of the diff.

After going back to review, I see the route evaluation order is covered already, but it feels like this matching description is in an inconvenient place as it comes after the route exit description instead of being part of the route evaluation explanation.

The design of the syntax is intended to convey a linear flow for routing, i.e. follow this `route` `when` these conditions match to this `exit`, otherwise try the next `route`, and fall through to the default `exit` if no more specific `route` matches.

Seems that the explanation would be more clear if it followed a similar linear order, rather than putting the route when condition matching after the route evaluation order and exit description.

Reply:
Ok, that makes sense. I will work on the overall order and clarify the individual conditions mentioned earlier.

Reply:
I pushed an update with some rewording and moving the exit section. here is how it looks now:
![image](https://github.com/aklivity/zilla-docs/assets/3012311/3562d2dd-1676-48ce-a1a8-596c5fe5f7b8)